plugins/catalog-backend/src/database/DefaultProcessingDatabase.ts

```diff
@@ -357,13 +356,11 @@ export class DefaultProcessingDatabase implements ProcessingDatabase {
         this.options.logger.warn(
           `Detected conflicting entityRef ${entityRef} already referenced by ${conflictingKey} and now also ${locationKey}`,
         );
-        conflictingStateReferences.push(entityRef);
       }
     }
 
     // Replace all references for the originating entity or source and then create new ones
     await tx<DbRefreshStateReferencesRow>('refresh_state_references')
-      .whereNotIn('target_entity_ref', conflictingStateReferences)
```

Comment:
As far as we can till this behavior was incorrect, and we updated the test. The references from entity E1 to E2 should not stick around if E1 is no longer the producer of E2